### PR TITLE
support customized dockerfile

### PR DESCRIPTION
**Pull request by supereagle**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR fixes** *(optional, close the issue(s) when PR gets merged)*:

Fixes #

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/worker/stage/stage.go [hunk 1]
```diff
@@ -305,16 +306,22 @@ func (sm *stageManager) ExecImageBuild(stage *api.ImageBuildStage) ([]string, er
 	}
 
 	builtImages := []string{}
+	cloneDir := scm.GetCloneDir()
 	for _, buildInfo := range stage.BuildInfos {
 		opt.Dockerfile = "Dockerfile"
-		if buildInfo.DockerfilePath != "" {
-			opt.Dockerfile = strings.TrimPrefix(strings.TrimPrefix(buildInfo.DockerfilePath, buildInfo.ContextDir), "/")
+		if buildInfo.Dockerfile != "" {
+			if err = osutil.ReplaceFile(cloneDir+"/Dockerfile", strings.NewReader(buildInfo.Dockerfile)); err != nil {
```
Comment (zhujian7) [inline pkg/worker/stage/stage.go]: `buildInfo.ContextDir/Dockerfile` maybe better.

Comment (supereagle) [reply pkg/worker/stage/stage.go]: Have fixed.

**Pull request merged**